tools/test-js-with-node

```diff
@@ -252,36 +266,51 @@ def run_tests_via_node_js() -> int:
     # after making sure tests will pass.
     node_tests_cmd = ["node", "--stack-trace-limit=100", INDEX_JS]
     if individual_files:
+        # If we passed a specific set of tests, run in serial mode.
+        global parallel
+        parallel = 1
         files = individual_files
     else:
         files = sorted(glob.glob(os.path.join(ROOT_DIR, "frontend_tests/node_tests/*.js")))
 
-    node_tests_cmd += clean_files(files)
-    if options.coverage:
-        coverage_dir = os.path.join(ROOT_DIR, "var/node-coverage")
-
-        nyc = os.path.join(ROOT_DIR, "node_modules/.bin/nyc")
-        command = [nyc, "--extension", ".hbs", "--extension", ".ts"]
-        command += ["--report-dir", coverage_dir]
-        command += ["--temp-directory", coverage_dir]
-        command += ["-r=lcov", "-r=json", "-r=text-summary"]
-        command += node_tests_cmd
-    else:
-        # Normal testing, no coverage analysis.
-        # Run the index.js test runner, which runs all the other tests.
-        command = node_tests_cmd
+    test_files = clean_files(files)
 
     print("Starting node tests...")
 
     # If we got this far, we can run the tests!
-    try:
-        ret = subprocess.check_call(command)
-    except OSError:
-        print(f"Bad command: {command}")
-        raise
-    except subprocess.CalledProcessError:
-        print("\n** Tests failed, PLEASE FIX! **\n")
-        sys.exit(1)
+    ret = 0
+    if parallel > 1:
```

Comment:
I believe you can refactor this to avoid fiddling with existing code and creating double nested if construct over here.

I can just have the
```
if parallel > 1:
    dosomething
    return ret
```

Above the existing code you deleted. This way you do different op for parellel task and exit early or you can keep executing the non parellel route. This will restore the big nested part in the else statement to its original state.